### PR TITLE
refactor table view into data analysis component

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

Refactor the table view into the data analysis component from the individual feature importances component.
Note there will be a follow-up PR to add a "mini-table" in the feature importances component to select a subset of indexes from the overall selected indexes in the tabular view.  Also, we will still show the table by default if the dataset explorer component is missing in the follow-up PR.

New data analysis view with new "Table view" tab:

image

New individual feature importances tab:

image

## Checklist

- [x] I have added screenshots above for all UI changes.
- [x] I have added e2e tests for all UI changes.
- [x] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+            label={localization.formatString(
+              localization.ModelAssessment.FeatureImportances.SelectionCounter,
+              this.selection.count,
+              this.maxSelectable
+            )}
+            calloutTitle={undefined}
+            renderOnNewLayer
+            type="label"
+          >
+            <Text block>
```
Comment (RubyZ10) [inline libs/dataset-explorer/src/lib/TableView/TableView.tsx:122]: Can we not set text with block? with block,  when making screen too small , the overlfow part of the text will not be seen.

Comment (RubyZ10) [reply libs/dataset-explorer/src/lib/TableView/TableView.tsx:122]: Could you please try to change the screen size to 320px? If the text is still visible, we should be fine

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                    columns={this.state.columns}
+                    groups={this.state.groups}
+                    setKey="set"
+                    layoutMode={DetailsListLayoutMode.fixedColumns}
+                    constrainMode={ConstrainMode.unconstrained}
+                    onRenderDetailsHeader={this.onRenderDetailsHeader}
+                    selectionPreservedOnEmptyClick
+                    ariaLabelForSelectionColumn="Toggle selection"
+                    checkButtonAriaLabel="Row checkbox"
+                    // checkButtonGroupAriaLabel="Group checkbox"
```
Comment (RubyZ10) [inline libs/dataset-explorer/src/lib/TableView/TableView.tsx:143]: remove?

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                    items={this.state.rows}
+                    columns={this.state.columns}
+                    groups={this.state.groups}
+                    setKey="set"
+                    layoutMode={DetailsListLayoutMode.fixedColumns}
+                    constrainMode={ConstrainMode.unconstrained}
+                    onRenderDetailsHeader={this.onRenderDetailsHeader}
+                    selectionPreservedOnEmptyClick
+                    ariaLabelForSelectionColumn="Toggle selection"
+                    checkButtonAriaLabel="Row checkbox"
```
Comment (RubyZ10) [inline libs/dataset-explorer/src/lib/TableView/TableView.tsx:142]: wondering if we need to localize aria label?

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    return (
+      <div>
+        {defaultRender?.({
+          ...props,
+          onRenderColumnHeaderTooltip,
+          selectAllVisibility: SelectAllVisibility.hidden
+        })}
+      </div>
+    );
+  };
```
Comment (RubyZ10) [inline libs/dataset-explorer/src/lib/TableView/TableView.tsx:265]: Could you please create onRenderDetailsHeader.tsx and move the component there? Thanks

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+  private _onRenderGroupHeader = (props?: IDetailsGroupDividerProps) => {
+    const classNames = tableViewStyles();
```
Comment (RubyZ10) [inline libs/dataset-explorer/src/lib/TableView/TableView.tsx:268]: Could you please create a onRenderGroupHeader.tsx and move this component there? Thanks!

**Review by RubyZ10 — APPROVED** [review]

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx (hunk 1 above)
Comment (imatiach-msft) [reply libs/dataset-explorer/src/lib/TableView/TableView.tsx:122]: removed block, thanks!

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx (hunk 2 above)
Comment (imatiach-msft) [reply libs/dataset-explorer/src/lib/TableView/TableView.tsx:143]: this was in the original file where I refactored the code from, but I removed it anyway

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx (hunk 4 above)
Comment (imatiach-msft) [reply libs/dataset-explorer/src/lib/TableView/TableView.tsx:265]: done!

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx (hunk 5 above)
Comment (imatiach-msft) [reply libs/dataset-explorer/src/lib/TableView/TableView.tsx:268]: done!

File: libs/dataset-explorer/src/lib/TableView/TableView.tsx (hunk 3 above)
Comment (imatiach-msft) [reply libs/dataset-explorer/src/lib/TableView/TableView.tsx:142]: done, localized just in case both aria labels

**Pull request merged**